### ISSUE TITLE
haproxy:latest (Docker) freezes when referencing stick table inside brackets

### ISSUE THREAD

**Issue by tplueer** [haproxy/haproxy#226]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 2.0.3 2019/07/23 - https://haproxy.org/
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = gcc
  CFLAGS  = -O2 -g -fno-strict-aliasing -Wdeclaration-after-statement -fwrapv -Wno-format-truncation -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-old-style-declaration -Wno-ignored-qualifiers -Wno-clobbered -Wno-missing-field-initializers -Wno-implicit-fallthrough -Wno-stringop-overflow -Wno-cast-function-type -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1

Feature list : +EPOLL -KQUEUE -MY_EPOLL -MY_SPLICE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED -REGPARM -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H -VSYSCALL +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -MY_ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with OpenSSL version : OpenSSL 1.1.1c  28 May 2019
Running on OpenSSL version : OpenSSL 1.1.1c  28 May 2019
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.3
Built with network namespace support.
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with PCRE2 version : 10.32 2018-09-10
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with the Prometheus exporter as a service

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTX        side=FE|BE     mux=H2
              h2 : mode=HTTP       side=FE        mux=H2
       <default> : mode=HTX        side=FE|BE     mux=H1
       <default> : mode=TCP|HTTP   side=FE|BE     mux=PASS

Available services :
	prometheus-exporter

Available filters :
	[SPOE] spoe
	[COMP] compression
	[CACHE] cache
	[TRACE] trace

```

```
Linux 7d3854220c4b 5.2.7-200.fc30.x86_64 #1 SMP Thu Aug 8 05:35:29 UTC 2019 x86_64 GNU/Linux
```
## What's the configuration?

```
global
        log stdout format raw local0 info  
        stats socket ipv4@0.0.0.0:9999 level admin
        stats timeout 30s
        daemon

	ssl-server-verify none
	ssl-default-bind-options no-sslv3
	ssl-default-bind-options no-tls-tickets
	ssl-default-bind-ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA
	tune.ssl.default-dh-param 2048

defaults
    log global
    option http-server-close
    option httplog
    mode http
    timeout http-request 5s
    timeout connect 5s
    timeout server 10s
    timeout client 30s

listen stats 
    bind :1936
    mode http
    stats enable
    stats hide-version
    stats realm Haproxy\ Statistics
    stats uri /stats
    http-request set-log-level silent


listen varnish_backend
    mode            http
    bind            0.0.0.0:8080
    balance         roundrobin
    option          httpclose
    option          forwardfor
    option          http-server-close
    option          httpchk HEAD / HTTP/1.0
    server backend_01 "${BACKEND_SERVER_IP}" ssl verify none check weight 1 maxconn 3000


frontend proxy_in
    # Listen
    bind ipv4@0.0.0.0:80
    bind ipv4@0.0.0.0:443 ssl crt /usr/local/etc/haproxy/ssl/cert.pem

    # FORBIDDEN USER_AGENTS
    acl FORBIDDEN_UA hdr_reg(User-Agent) -i -f /usr/local/etc/haproxy/blacklists/badbots.txt

    # FORBIDDEN REFERRER
    acl FORBIDDEN_REFERRER hdr_sub(Referer) -i -f /usr/local/etc/haproxy/blacklists/badref.txt

    # FORBIDDEN_IP's
    acl FORBIDDEN_IP src -f /usr/local/etc/haproxy/blacklists/blacklist.txt
    acl FORBIDDEN_IP src -f /usr/local/etc/haproxy/blacklists/tor-exit-nodes.txt
    acl FORBIDDEN_IP src -f /usr/local/etc/haproxy/blacklists/scammers.txt
    acl FORBIDDEN_IP src -f /usr/local/etc/haproxy/blacklists/other-scammers.txt
    acl FORBIDDEN_IP src -f /usr/local/etc/haproxy/blacklists/exploited-servers.txt
    acl FORBIDDEN_IP src -f /usr/local/etc/haproxy/blacklists/proxies.txt

    # ALLOWED_IP's
    acl ALLOWED_IP src -f /usr/local/etc/haproxy/whitelists/whitelist.txt

    # Stick Table
    stick-table type ip size 500k expire 5m store conn_cur,conn_rate(3s),http_req_rate(5s),http_err_rate(10s)

    # Track IP
    tcp-request connection track-sc0 src

    # Caching Files
    acl static_content  path_end    -i  .jpg .png .gif .css .js .ttf .woff .woff2 .gif .doc .csv .jpeg .pdf .bmp .ico .ppt .xls .docx .svg .webp

    # MAX CONN
    maxconn 3000

    # Filters
    acl FILTER_CONN_CUR always_true    # Filter based on current connections
    acl FILTER_CONN_RATE always_true   # Filter based on tcp connection rate
    acl FILTER_REQ_RATE always_true    # Filter based on http req rate
    acl FILTER_ERROR_RATE always_true  # Filter based on error rate
    acl FILTER_USER_AGENT always_true  # Filter based on user agents
    acl FILTER_BAD_REQUEST always_true # Filter based on bad requests
    acl FILTER_BAD_REFERRER always_true # Filter based on bad referrer

    # Bot Protection
    acl FILTER_JAVASCRIPT always_true

    # Proxy Features
    acl FEATURE_CACHING always_true     # Enable static content caching via varnish

    # FORBIDDEN_HDR's
    acl FORBIDDEN_HDR hdr_cnt(host) gt 1
    acl FORBIDDEN_HDR hdr_cnt(content-length) gt 1
    acl FORBIDDEN_HDR hdr_val(content-length) lt 0
    acl FORBIDDEN_HDR hdr_cnt(proxy-authorization) gt 0
    acl FORBIDDEN_HDR hdr_cnt(x-xsrf-token) gt 1
    acl FORBIDDEN_HDR hdr_len(x-xsrf-token) gt 36

    # FORBIDDEN_URI's
    acl FORBIDDEN_URI url_reg -i .*(\.|%2e)(\.|%2e)(%2f|%5c|/|\\\\)
    acl FORBIDDEN_URI url_sub -i %00 <script xmlrpc.php
    acl FORBIDDEN_URI path_beg /_search /_nodes
    acl FORBIDDEN_URI path_end -i .ida .asp .dll .exe .sh .pl .py .so
    acl FORBIDDEN_URI path_dir -i chat phpbb sumthin horde _vti_bin MSOffice

    # REQUEST_TO_BIG
    acl REQUEST_TOO_BIG hdr_val(content-length) gt 20000

    # RANGE_HEADERS
    acl WEIRD_RANGE_HEADERS hdr_cnt(Range) gt 10

    # CONN_CUR
    http-request tarpit if { sc0_conn_cur ge 10 } FILTER_CONN_CUR !ALLOWED_IP

    # CONN_RATE
    http-request tarpit if { sc0_conn_rate ge 20 } FILTER_CONN_RATE !ALLOWED_IP

    # REQ_RATE
    http-request tarpit if { sc0_http_req_rate ge 100 } FILTER_REQ_RATE !ALLOWED_IP 

    # ERROR_RATE
    http-request tarpit if { sc0_http_err_rate ge 20 } FILTER_ERROR_RATE !ALLOWED_IP

    # BAD_REQUESTS
    http-request tarpit if FORBIDDEN_HDR FILTER_BAD_REQUEST !ALLOWED_IP
    http-request tarpit if FORBIDDEN_URI FILTER_BAD_REQUEST !ALLOWED_IP

    # IS_BOT
    http-request tarpit if { src_get_gpt0(bot_detection) eq 1 } !ALLOWED_IP 
 
    # USER_AGENT
    http-request deny if FORBIDDEN_UA FILTER_USER_AGENT !ALLOWED_IP

    # BAD_REFERRER
    http-request deny if FORBIDDEN_REFERRER FILTER_BAD_REFERRER !ALLOWED_IP

    # X-FORWARDED-FOR HEADER
    option forwardfor

    http-request add-header X-Forwarded-Proto https if { ssl_fc } 
    http-request add-header X-Forwarded-Proto http if !{ ssl_fc } 
    
    # CACHING BACKEND
    use_backend varnish if FEATURE_CACHING static_content
    
    # BOT DETECTION
    use_backend bot_detection if { src_get_gpc0(bot_detection) eq 0 } FILTER_JAVASCRIPT
    
    # DEFAULT
    default_backend proxy_out

backend bot_detection
     stick-table type ip size 500k expire 7d store gpt0,gpc0
     tcp-request content track-sc1 src
     server bot_detection anti_bot:80 check weight 1 maxconn 3000

backend proxy_out
    server backend_01 "${BACKEND_SERVER_IP}" ssl verify none check weight 1 maxconn 3000

backend varnish
	server varnish_01 varnish:80 check weight 1 maxconn 3000

```

## Steps to reproduce the behavior

1. Download haproxy latest docker container 
2. Use the config
3. Try to start the container or run haproxy -c -f /usr/local/etc/haproxy/haproxy.cfg

## Actual behavior

The haproxy process freezes and no logs are sent to stdout also the config check freezes and does not provide any output

## Expected behavior

The container starts normally

## Do you have any idea what may have caused this?
No but i found out that if you remove the table inside the brackets or comment out the following lines the container starts

**Dont't works:**
```
http-request tarpit if { src_get_gpt0(bot_detection) eq 1 } !ALLOWED_IP 
use_backend bot_detection if { src_get_gpc0(bot_detection) eq 0 } FILTER_JAVASCRIPT

```
**Works:**
```
http-request tarpit if { src_get_gpt0() eq 1 } !ALLOWED_IP 
use_backend bot_detection if { src_get_gpc0() eq 0 } FILTER_JAVASCRIPT
```
## Do you have an idea how to solve the issue?
Unfortunately not

**Comment by wtarreau** [comment]
A similar issue was fixed in 2.0.5, please update it and try again.

**Comment by tplueer** [comment]
After upgrading to 2.0.5 it works! Many thanks!
